torchdata/dataloader2/dataloader2.py

```diff
@@ -206,9 +212,14 @@ def __iter__(self) -> DataLoader2Iterator[T_co]:
                 if self._reset_seed:
                     self._seed_generator.seed(self._seed)
                     self._reset_seed = False
```

Comment:
In this branch of logic, we have `self._seed` and we use it to set. Should we set `self._seed = None` afterwards?
Currently, it is relying on the fact that `self._reset_seed == False` until `self.seed()` method is called again.